src/app/medicine/medicine.cpp

```diff
@@ -0,0 +1,8 @@
+#include "medicine.hpp"
+
+Medicine::Medicine(const std::string &name, double price, int amount, int expiration_date,
+                   std::pair<double, double> storage_temperature, std::vector<std::string> composition)
+    : Product(name, price, amount), expiration_date_(expiration_date), storage_temperature_(storage_temperature),
```

Comment:
Think about adding private function for expiration date validation. We don't want to add medicine that is already expired.

Reply:
Function added with // TODO comment (there is no data type Date currently, so writing proper function is in this moment impossible, but will be possible in future).